File: strategy-game-frontend/src/core/models/province.ts
import {ResourceType} from "./resourceType";
import {TradeRoute} from "./tradeRoute";

export interface Province {
    provinceId: string,
    countryId: string,
    cityIds: string[],
    provinceCapitalCityId: string,
    resources: Map<ResourceType, number> | null
    tradeRoutes: TradeRoute[]
}

File: strategy-game-backend/src/main/kotlin/de/ruegnerlukas/strategygame/backend/gameengine/ports/models/Province.kt
package de.ruegnerlukas.strategygame.backend.gameengine.ports.models

import de.ruegnerlukas.strategygame.backend.common.models.resources.ResourceCollection

data class Province(
    val provinceId: String,
    val countryId: String,
    val cityIds: MutableList<String>,
    val provinceCapitalCityId: String,

    var resourcesProducedPrevTurn: ResourceCollection = ResourceCollection.basic(),
    var resourcesProducedCurrTurn: ResourceCollection = ResourceCollection.basic(),
    var resourcesConsumedCurrTurn: ResourceCollection = ResourceCollection.basic(),
    var resourcesMissing: ResourceCollection = ResourceCollection.basic(),
)
